packages/wrangler/src/__tests__/type-generation.test.ts

```diff
@@ -255,4 +325,220 @@ describe("generateTypes()", () => {
 		"
 	`);
 	});
+
+	it("should include secrets from .dev.vars", async () => {
+		fs.writeFileSync(
+			"./wrangler.toml",
+			TOML.stringify({
+				vars: {
+					myTomlVarA: "A from wrangler toml",
+					myTomlVarB: "B from wrangler toml",
+				},
+			} as TOML.JsonMap),
+			"utf-8"
+		);
+
+		const localVarsEnvContent = dedent`
+		# Preceding comment
+		SECRET_A="A from .dev.vars"
+		MULTI_LINE_SECRET="A: line 1
+		line 2"
+		UNQUOTED_SECRET= unquoted value
+		`;
+		fs.writeFileSync(".dev.vars", localVarsEnvContent, "utf8");
+
+		await runWrangler("types");
+
+		expect(std.out).toMatchInlineSnapshot(`
+		"interface Env {
+			myTomlVarA: \\"A from wrangler toml\\";
+			myTomlVarB: \\"B from wrangler toml\\";
+			SECRET_A: string;
+			MULTI_LINE_SECRET: string;
+			UNQUOTED_SECRET: string;
+		}
+		"
+	`);
+	});
+
+	it("should override vars with secrets", async () => {
+		fs.writeFileSync(
+			"./wrangler.toml",
+			TOML.stringify({
+				vars: {
+					MY_VARIABLE: "my variable",
```

Comment:
Perhaps make this var a complex object instead of a string, so that we can see that it really does get overridden?

```suggestion
					MY_VARIABLE: { NAME = "foo" },
```

Reply:
mh.... but instead of changing it I'd add this test if it's ok by you 🙂

I was testing the overriding because `wrangler types`, when it encounters a var string binding, it types is as a const string (e.g. `MY_VARIABLE: "my variable"` instead of `MY_VARIABLE: string`)

and what I wanted to test here is that when `wrangler types` encounters the same string var binding in both the toml and `.dev.vars` it goes with the generic `string` alternative instead of picking the const one

Reply:
Ah I see. I didn't spot that.